docs/dev/Lua API.rst

```diff
@@ -5766,82 +5766,134 @@ The names of the functions are also available as the keys of the
+
+  Animate workshop by replacing displayed tiles (or graphical tiles). There are two ways this works:
+  if ``frame_skip>=0`` then it shows each frame for ``frame_skip`` of frames or if ``frame_skip<0``
+  Frames are synchronized to the machine this building is connected to.
+
+  :workshop_type:   custom workshop string id e.g. ``SOAPMAKER`` or numeric id
+  :frames:          table of frames. Each frame is sparse flat table with ids from ``0`` to ``31*31-1``.
+                    Each frame tile is table of integers from 4 to 8 members long. Tile members are as
+                    follow: ``tile``, ``foreground color``, ``background color``, ``bright``,
+                    ``graphics tile``, ``overlay tile``, ``signpost tile``, ``item tile``.
+                    First 4 are function same as ascii workshop definition. The latter 4 are graphics
+                    layers. ``signpost tile`` is an optional row that sticks up over the workshop.
 
-    :name:
-        custom workshop id e.g. ``SOAPMAKER``
+  :frame_skip:      How many ticks to display one frame. If set to negative number (or skipped) frames
+                    are synchronized with machine animation.
 
-        .. note:: this is the only mandatory field.
+* ``setAnimationInfoAuto(workshop_type, make_graphics_too, [frame_length], [gear_tiles])``
 
-    :fix_impassible:
-        if true make impassable tiles impassable to liquids too
-    :consume:
-        how much machine power is needed to work.
-        Disables reactions if not supplied enough and ``needs_power==1``
-    :produce:
-        how much machine power is produced.
-    :needs_power:
-        if produced in network < consumed stop working, default true
-    :gears:
-        a table or ``{x=?,y=?}`` of connection points for machines.
-    :action:
-        a table of number (how much ticks to skip) and a function which
-        gets called on shop update
-    :animate:
-        a table of frames which can be a table of:
+  Animate workshop as with function above but generate frames automatically. This works by finding
+  tiles which have gears and animating them with alternating gear tiles.
 
-        a. tables of 4 numbers ``{tile,fore,back,bright}`` OR
-        b. empty table (tile not modified) OR
-        c. ``{x=<number> y=<number> + 4 numbers like in first case}``,
-           this generates full frame useful for animations that change little (1-2 tiles)
+  :workshop_type:     custom workshop string id e.g. ``SOAPMAKER`` or numeric id
+  :make_graphics_too:  replace same tiles in graphics mode with tiles from vanilla df mechanism
+  :frame_length:      How many ticks to display one frame. If set to negative number (or skipped) frames
+                      are synchronized with machine animation.
+  :gear_tiles:        Optional array of 2 or 4 indexes. First two define ascii tiles and next two graphics tiles.
+                      This overrides default gear tiles.
```

Comment:
a map might be more instructive instead of a variable length list, e.g.:
```lua
local bhacks = require('plugins.building-hacks')
bhacks.setAnimationInfoAuto('DRAGON_ENGINE', true, 4, {
    gear_ch=42, -- looks for this in ascii
    gear_ch_alt=15,  -- replaces with this for animated frames
    gear_tile=dfhack.screen.findGraphicsTile('WARMIST_DRAGON_ENGINE', 1, 0),
    gear_tile_alt=dfhack.screen.findGraphicsTile('WARMIST_DRAGON_ENGINE', 4, 0),
})`